[rocdecode] rocDecode utility class documentation (#4475)
ROCm/rocm-systems: projects/rocdecode/docs/reference/rocDecode-demux.rst

.. meta::
  :description: The rocDecode VideoDemuxer utility class
  :keywords: video demuxing, video demultiplexing, rocDecode, core APIs, AMD, ROCm, utility class

********************************************************************

> The rocDecode VideoDemuxer utility class

Reviewer: ```suggestion
rocDecode VideoDemuxer utility class
```

Author: I like the the

********************************************************************

> The rocDecode FFMpeg demuxer utility class, `VideoDemuxer <https://github.com/ROCm/rocm-systems/tree/develop/projects/rocdecode/utils/video_demuxer.h>`_, is used to demultiplex (demux) frames that can then be consumer by the :doc:`FFMpeg decoder <./rocDecode-ffmpeg-decoder>`. 

Reviewer: ```suggestion
The rocDecode FFMpeg demuxer utility class, `VideoDemuxer <https://github.com/ROCm/rocm-systems/tree/develop/projects/rocdecode/utils/video_demuxer.h>`_, is used to demultiplex (demux) frames that can then be consumed by the :doc:`FFMpeg decoder <./rocDecode-ffmpeg-decoder>`. 
```

> 

Reviewer 2: The consumer of VideoDemuxer's output (the coded frames) can be either the SW decoder (FFMPEG) or the HW decoder (the Roc decoder). Suggest to also add link of Roc decoder here.

The ``VideoDemuxer()`` constructor takes either a file path or a StreamProvider object. The StreamProvider class is an abstract class that feeds a stream of packets to the demuxer. The StreamProvider ``GetData()`` and ``GetBufferSize()`` functions must be implemented to use a StreamProvider.

The VideoDemuxer ``Demux()`` function extracts the next video packet from the stream, and returns a pointer to the packet data, the packet size, and optionally the presentation timestamp. 

> ``Demux()`` demuxes frames sequentially starting at the beginning of the stream. The ``seek()`` function is used to start demuxing from a different frame.

Reviewer: ```suggestion
``Demux()`` demuxes frames sequentially, starting at the beginning of the stream. The ``seek()`` function is used to start demuxing from a different frame.
```

> A ``VideoSeekContext`` is passed to ``seek()``. The seek context that specifies a seek criteria and a seek mode. The seek criteria describes whether the demuxer needs to seek to a specific frame or seek to a specific timestamp. The seek mode indicates whether the demuxer should seek to the exact frame or to the previous keyframe. 

Reviewer: ```suggestion
A ``VideoSeekContext`` is passed to ``seek()``. The seek context specifies a seek criteria and a seek mode. The seek criteria describe whether the demuxer needs to seek to a specific frame or seek to a specific timestamp. The seek mode indicates whether the demuxer should seek to the exact frame or to the previous keyframe. 
```

Reviewer: criteria or criterium

Author: The seek criteria is an object, so it describes
